src/api.rs

```diff
@@ -85,7 +91,7 @@ unsafe fn join_inject<A,B,RA,RB>(oper_a: A,
     let mut latch_b = LockLatch::new();
     let mut job_b = Job::new(&mut code_b, &mut latch_b);
 
-    thread_pool::get_registry().inject(&[&mut job_a, &mut job_b]);
+    thread_pool::get_registry(num_cpus::get()).inject(&[&mut job_a, &mut job_b]);
```

Comment:
(If we had an easy way to specify the default value, I'd prefer that we use that here rather than hardcode `num_cpus::get()` -- or rather, this seems like the wrong place to hardcode that, I'd prefer it's hardcoded in the `get_registry` routine itself, which seems more central.)